analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CloudNative/DurableEntityInterfaceRestrictions.cs

```diff
@@ -0,0 +1,216 @@
+        await client.SignalEntityAsync<IInheritsInvalid>(id, x => { });         // Noncompliant {{Use valid entity interface. IInheritsInvalid contains method "Method" with 2 parameters. Zero or one are allowed.}}
+        await client.SignalEntityAsync<IInvalid>(id, x => { });                 // Noncompliant {{Use valid entity interface. IInvalid contains method "Method" with 2 parameters. Zero or one are allowed.}}
+        await client.SignalEntityAsync<IMoreArguments>(id, x => { });           // Noncompliant {{Use valid entity interface. IMoreArguments contains method "Method" with 2 parameters. Zero or one are allowed.}}
+        await client.SignalEntityAsync<IReturnsInt>(id, x => { });              // Noncompliant {{Use valid entity interface. IReturnsInt contains method "Method" with invalid return type. Only "void", "Task" and "Task<T>" are allowed.}}
+        await client.SignalEntityAsync<IReturnsTaskArray>(id, x => { });        // Noncompliant {{Use valid entity interface. IReturnsTaskArray contains method "Method" with invalid return type. Only "void", "Task" and "Task<T>" are allowed.}}
+        await client.SignalEntityAsync<IReturnsObject>(id, x => { });           // Noncompliant {{Use valid entity interface. IReturnsObject contains method "Method" with invalid return type. Only "void", "Task" and "Task<T>" are allowed.}}
+        await client.SignalEntityAsync<IGenericInterface<int>>(id, x => { });   // Noncompliant {{Use valid entity interface. IGenericInterface is generic.}}
+        await client.SignalEntityAsync<IGenericMethod>(id, x => { });           // Noncompliant {{Use valid entity interface. IGenericMethod contains generic method "Method".}}
+        await client.SignalEntityAsync<IProperty>(id, x => { });                // Noncompliant {{Use valid entity interface. IProperty contains property "Value". Only methods are allowed.}}
+        await client.SignalEntityAsync<IIndexer>(id, x => { });                 // Noncompliant {{Use valid entity interface. IIndexer contains property "this[]". Only methods are allowed.}}
+        await client.SignalEntityAsync<IEvent>(id, x => { });                   // Noncompliant {{Use valid entity interface. IEvent contains event "Event". Only methods are allowed.}}
+        await client.SignalEntityAsync<IEvent>(id, x => { });                   // Noncompliant {{Use valid entity interface. IEvent contains event "Event". Only methods are allowed.}}
+        await client.SignalEntityAsync<NotInterfaceClass>(id, x => { });        // Noncompliant {{Use valid entity interface. NotInterfaceClass is not an interface.}}
+        await client.SignalEntityAsync<NotInterfaceStruct>(id, x => { });       // Noncompliant {{Use valid entity interface. NotInterfaceStruct is not an interface.}}
+    }
+
+    public async Task FromDurableClient(IDurableClient inheritedClient)
+    {
+        await inheritedClient.SignalEntityAsync<IInvalid>(id, x => { });                 // Noncompliant
+    }
+}
```

Comment:
Type parameter test is missing:

```suggestion
    public async Task WithTypeParameter<T>()
    {
        await client.SignalEntityAsync<T>(id, x => { });                 // Compliant.
    }
}
```